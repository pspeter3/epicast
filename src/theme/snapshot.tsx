import * as React from "react";
import TestRenderer from "react-test-renderer";

export const snapshot = (name: string, SFC: React.SFC<{}>): void => {
    describe(name, () => {
        it(`should match snapshot`, () => {
            const component = TestRenderer.create(<SFC>Snapshot</SFC>);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
};
