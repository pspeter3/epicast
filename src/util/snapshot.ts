import { shallow } from "enzyme";
import * as React from "react";

export type Factory = () => React.ReactElement<any>;

export const snapshot = (factory: Factory): (() => void) => {
    return () => {
        expect(shallow(factory())).toMatchSnapshot();
    };
};

export const snapshotSuite = (name: string, factory: Factory): void => {
    describe(name, () => {
        it("should match snapshot", snapshot(factory));
    });
};
