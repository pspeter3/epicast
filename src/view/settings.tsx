import * as React from "react";
import { BottomButton, IconButton, SubHeaderButton } from "../theme/buttons";
import { MinusCircleIcon, PlusCircleIcon } from "../theme/icons";
import { NumericInput } from "../theme/inputs";
import { Main, Row, Section } from "../theme/layout";
import { SubHeader, Text } from "../theme/typography";

export const Settings: React.SFC<{}> = () => {
    return (
        <React.Fragment>
            <Section>
                <Row>
                    <Text>Played Deck Size</Text>
                    <IconButton>
                        <MinusCircleIcon />
                    </IconButton>
                    <NumericInput value={60} />
                    <IconButton>
                        <PlusCircleIcon />
                    </IconButton>
                </Row>
                <Row>
                    <Text>Epidemic Count</Text>
                    <IconButton>
                        <MinusCircleIcon />
                    </IconButton>
                    <NumericInput value={6} />
                    <IconButton>
                        <PlusCircleIcon />
                    </IconButton>
                </Row>
            </Section>
            <Row>
                <SubHeader>Infection Deck</SubHeader>
                <SubHeaderButton>Add City</SubHeaderButton>
            </Row>
            <Main>
                <Section>
                    <Row>
                        <Text>San Francisco</Text>
                        <IconButton>
                            <MinusCircleIcon />
                        </IconButton>
                        <NumericInput value={2} />
                        <IconButton>
                            <PlusCircleIcon />
                        </IconButton>
                    </Row>
                </Section>
            </Main>
            <BottomButton>Save</BottomButton>
        </React.Fragment>
    );
};
