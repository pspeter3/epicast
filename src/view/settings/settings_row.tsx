import * as React from "react";
import { IconButton } from "../../theme/buttons";
import { MinusCircleIcon, PlusCircleIcon } from "../../theme/icons";
import { NumericInput } from "../../theme/inputs";
import { Row } from "../../theme/layout";
import { Text } from "../../theme/typography";

export interface Props {
    name: string;
    displayName?: string;
    value: number;
    onDecrement: (name: string) => void;
    onChange: (name: string, value: number) => void;
    onIncrement: (name: string) => void;
}

export class SettingsRow extends React.PureComponent<Props, {}> {
    public render() {
        const { name, displayName, value } = this.props;
        return (
            <Row>
                <Text>{displayName || name}</Text>
                <IconButton onClick={this._onDecrement}>
                    <MinusCircleIcon />
                </IconButton>
                <NumericInput onChange={this._onChange} value={value} />
                <IconButton onClick={this._onIncrement}>
                    <PlusCircleIcon />
                </IconButton>
            </Row>
        );
    }

    private _onDecrement = () => {
        this.props.onDecrement(this.props.name);
    };

    private _onChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
        const value = evt.target.value;
        if (value) {
            this.props.onChange(this.props.name, parseInt(evt.target.value, 10));
        }
    };

    private _onIncrement = () => {
        this.props.onIncrement(this.props.name);
    };
}
