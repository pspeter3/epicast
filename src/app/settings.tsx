import * as React from "react";
import { Config } from "../core/types";
import { ActionProps, Appbar } from "../theme/appbar";
import { classNames, focusClass } from "../theme/css";
import { FormField } from "../theme/form_field";
import { CodeIcon, SaveIcon } from "../theme/icons";
import {
    AlignItems,
    BackgroundColor,
    BorderColor,
    BorderRadius,
    BorderSize,
    BorderStyle,
    BoxShadow,
    Display,
    FontWeight,
    JustifyContent,
    Outline,
    Padding,
    Sizing,
    TextColor,
    TextDecoration,
    TextSize,
    Tracking,
} from "../theme/tailwind";
import { DialogService } from "../util/services";
import { difference, unit } from "../util/stacks";
import { CityRow } from "./city_row";
import { Routes } from "./routes";

export interface Props {
    config: Config;
    onConfigure: (config: Config) => boolean;
    services: {
        dialog: DialogService;
    };
}

export class Settings extends React.PureComponent<Props, Config> {
    public static displayName = "Settings";

    public static Legend: React.SFC<React.HTMLProps<HTMLLegendElement>> = props => (
        <legend
            {...props}
            className={classNames(
                FontWeight.Medium,
                Padding.A3,
                Sizing.H12,
                TextColor.Grey,
                TextSize.Small,
                Tracking.Wide,
                props.className,
            )}
        />
    );

    public static ButtonField: React.SFC<React.HTMLProps<HTMLButtonElement>> = props => (
        <div className={classNames(Padding.X3, Padding.Y1)}>
            <button
                {...props}
                className={classNames(
                    AlignItems.Center,
                    BorderColor.Base,
                    BorderRadius.Small,
                    BorderSize.A2,
                    BorderStyle.Solid,
                    Display.InlineFlex,
                    FontWeight.Medium,
                    JustifyContent.Center,
                    Padding.X4,
                    Sizing.H10,
                    Sizing.WFull,
                    TextColor.Base,
                    TextDecoration.Uppercase,
                    TextSize.Small,
                    focusClass(BackgroundColor.Lightest),
                    focusClass(BoxShadow.Large),
                    focusClass(Outline.None),
                    props.className,
                )}
            />
        </div>
    );

    public state = this.props.config;

    private _actions: ActionProps[] = [
        { icon: CodeIcon, href: Routes.Debug },
        {
            icon: SaveIcon,
            href: Routes.Dashboard,
            onClick: evt => {
                if (!this.props.onConfigure(this.state)) {
                    evt.preventDefault();
                }
            },
        },
    ];

    public render() {
        return (
            <>
                <Appbar actions={this._actions} />
                <main className={classNames(Padding.X1)}>
                    <FormField
                        label="Player Deck Size"
                        value={this.state.cards}
                        onChange={this._onChangeCards}
                    />
                    <FormField
                        label="Epidemic Count"
                        value={this.state.epidemics}
                        onChange={this._onChangeEpidemics}
                    />
                    <fieldset>
                        <Settings.Legend>Cities</Settings.Legend>
                        {Object.keys(this.state.cities).map(city => (
                            <CityRow
                                city={city}
                                value={this.state.cities[city]}
                                onChange={this._onChange}
                            />
                        ))}
                        <Settings.ButtonField onClick={this._onAddCity}>
                            Add City
                        </Settings.ButtonField>
                    </fieldset>
                </main>
            </>
        );
    }

    private _onChangeCards = (value: number) => {
        this.setState({
            ...this.state,
            cards: value,
        });
    };

    private _onChangeEpidemics = (value: number) => {
        this.setState({
            ...this.state,
            epidemics: value,
        });
    };

    private _onAddCity = () => {
        const city = this.props.services.dialog.prompt("City name");
        if (city === null) {
            return this.props.services.dialog.alert("Cannot set empty city name");
        }
        this.setState({
            ...this.state,
            cities: {
                ...this.state.cities,
                ...unit(city),
            },
        });
    };

    private _onChange = (city: string, value: number): void => {
        const cities =
            value > 0
                ? { ...this.state.cities, [city]: value }
                : difference(this.state.cities, { [city]: this.state.cities[city] });
        this.setState({
            ...this.state,
            cities,
        });
    };
}

Settings.Legend.displayName = "Settings.Legend";
Settings.ButtonField.displayName = "Settings.ButtonField";
