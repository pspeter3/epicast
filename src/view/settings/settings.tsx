import * as React from "react";
import { Redirect } from "react-router";
import { Config } from "../../core/types";
import { BottomButton, SubHeaderButton } from "../../theme/buttons";
import { Main, Row } from "../../theme/layout";
import { SubHeader } from "../../theme/typography";
import { DialogService } from "../../util/services";
import { difference, union, unit } from "../../util/stacks";
import { Routes } from "../routes";
import { SettingsRow } from "./settings_row";

export interface Services {
    dialog: DialogService;
}

export interface Props {
    config: Config;
    onConfigure: (config: Config) => void;
    services: Services;
}

export interface State {
    config: Config;
    submitted: boolean;
}

type Counter = "cards" | "epidemics";

export class Settings extends React.PureComponent<Props, State> {
    public state = {
        config: this.props.config,
        submitted: false,
    };

    public render() {
        const { config } = this.state;
        const { cities } = config;
        if (this.state.submitted) {
            return <Redirect to={Routes.Dashboard} />;
        }
        return (
            <React.Fragment>
                <SettingsRow
                    name="cards"
                    displayName="Player Deck Size"
                    value={config.cards}
                    onDecrement={this._onDecrementCounter}
                    onChange={this._onSetCounter}
                    onIncrement={this._onIncrementCounter}
                />
                <SettingsRow
                    name="epidemics"
                    displayName="Epidemic Count"
                    value={config.epidemics}
                    onDecrement={this._onDecrementCounter}
                    onChange={this._onSetCounter}
                    onIncrement={this._onIncrementCounter}
                />
                <Row>
                    <SubHeader>Infection Deck</SubHeader>
                    <SubHeaderButton onClick={this._onAddCity}>Add City</SubHeaderButton>
                </Row>
                <Main>
                    {Object.keys(cities).map(city => {
                        return (
                            <SettingsRow
                                key={city}
                                name={city}
                                value={cities[city]}
                                onDecrement={this._onDecrementCity}
                                onChange={this._onSetCity}
                                onIncrement={this._onIncrementCity}
                            />
                        );
                    })}
                </Main>
                <BottomButton onClick={this._onSave}>Save</BottomButton>
            </React.Fragment>
        );
    }

    private _onDecrementCounter = (name: string) => {
        this.setState({
            config: {
                ...this.state.config,
                [name]: this.state.config[name as Counter] - 1,
            },
        });
    };

    private _onSetCounter = (name: string, value: number) => {
        this.setState({
            config: {
                ...this.state.config,
                [name]: value,
            },
        });
    };

    private _onIncrementCounter = (name: string) => {
        this.setState({
            config: {
                ...this.state.config,
                [name]: this.state.config[name as Counter] + 1,
            },
        });
    };

    private _onAddCity = () => {
        const city = this.props.services.dialog.prompt("City Name");
        if (city !== null) {
            this._onIncrementCity(city);
        }
    };

    private _onDecrementCity = (city: string) => {
        const { config } = this.state;
        this.setState({
            config: {
                ...config,
                cities: difference(config.cities, unit(city)),
            },
        });
    };

    private _onSetCity = (city: string, value: number) => {
        const { config } = this.state;
        this.setState({
            config: {
                ...config,
                cities: {
                    ...config.cities,
                    [city]: value,
                },
            },
        });
    };

    private _onIncrementCity = (city: string) => {
        const { config } = this.state;
        this.setState({
            config: {
                ...config,
                cities: union(config.cities, unit(city)),
            },
        });
    };

    private _onSave = () => {
        const { config } = this.state;
        if (
            config.cards <= 0 ||
            config.epidemics <= 0 ||
            config.cards < config.epidemics ||
            Object.keys(config.cities).length === 0
        ) {
            this.props.services.dialog.alert("Invalid State");
            return;
        }
        if (this.props.services.dialog.confirm("Reset Game State?")) {
            this.props.onConfigure(this.state.config);
            this.setState({
                submitted: true,
            });
        }
    };
}
