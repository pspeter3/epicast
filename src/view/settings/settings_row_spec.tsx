import { shallow } from "enzyme";
import * as React from "react";
import { IconButton } from "../../theme/buttons";
import { NumericInput } from "../../theme/inputs";
import { snapshot } from "../../util/snapshot";
import { SettingsRow } from "./settings_row";

describe("SettingsRow", () => {
    it(
        "should match snapshot with name",
        snapshot(() => (
            <SettingsRow
                name="name"
                value={1}
                onDecrement={jest.fn()}
                onChange={jest.fn()}
                onIncrement={jest.fn()}
            />
        )),
    );

    it(
        "should match snapshot with displayName",
        snapshot(() => (
            <SettingsRow
                name="name"
                displayName="Display Name"
                value={1}
                onDecrement={jest.fn()}
                onChange={jest.fn()}
                onIncrement={jest.fn()}
            />
        )),
    );

    it("should handle decrement", () => {
        const name = "name";
        const onDecrement = jest.fn();
        const tree = shallow(
            <SettingsRow
                name={name}
                value={1}
                onDecrement={onDecrement}
                onChange={jest.fn()}
                onIncrement={jest.fn()}
            />,
        );
        tree
            .find(IconButton)
            .first()
            .simulate("click");
        expect(onDecrement).toHaveBeenCalledWith(name);
    });

    it("should handle change", () => {
        const name = "name";
        const value = 3;
        const onChange = jest.fn();
        const tree = shallow(
            <SettingsRow
                name={name}
                value={1}
                onDecrement={jest.fn()}
                onChange={onChange}
                onIncrement={jest.fn()}
            />,
        );
        tree.find(NumericInput).simulate("change", { target: { value: value.toString() } });
        expect(onChange).toHaveBeenCalledWith(name, value);
    });

    it("should handle empthy change", () => {
        const name = "name";
        const onChange = jest.fn();
        const tree = shallow(
            <SettingsRow
                name={name}
                value={1}
                onDecrement={jest.fn()}
                onChange={onChange}
                onIncrement={jest.fn()}
            />,
        );
        tree.find(NumericInput).simulate("change", { target: { value: "" } });
        expect(onChange).not.toHaveBeenCalled();
    });

    it("should handle increment", () => {
        const name = "name";
        const onIncrement = jest.fn();
        const tree = shallow(
            <SettingsRow
                name={name}
                value={1}
                onDecrement={jest.fn()}
                onChange={jest.fn()}
                onIncrement={onIncrement}
            />,
        );
        tree
            .find(IconButton)
            .last()
            .simulate("click");
        expect(onIncrement).toHaveBeenCalledWith(name);
    });
});
