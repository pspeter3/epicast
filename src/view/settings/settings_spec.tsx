import { shallow } from "enzyme";
import * as React from "react";
import { BottomButton, SubHeaderButton } from "../../theme/buttons";
import { snapshot } from "../../util/snapshot";
import { Settings } from "./settings";

describe("Settings", () => {
    const createProps = () => {
        return {
            config: {
                cards: 0,
                cities: {},
                epidemics: 0,
            },
            onConfigure: jest.fn(),
            services: {
                dialog: {
                    alert: jest.fn(),
                    confirm: jest.fn(),
                    prompt: jest.fn(),
                },
            },
        };
    };

    it("should match snapshot", snapshot(() => <Settings {...createProps()} />));

    describe("counters", () => {
        const suite = (name: string, displayName: string) => {
            describe(name, () => {
                it("should handle displayName", () => {
                    const props = createProps();
                    const tree = shallow(<Settings {...props} />);
                    const row = tree.find({ name });
                    expect(row.prop("displayName")).toBe(displayName);
                });

                it("should handle decrement", () => {
                    const props = createProps();
                    const tree = shallow(<Settings {...props} />);
                    const row = tree.find({ name });
                    row.prop("onDecrement")(name);
                    expect(tree.state().config[name]).toBe(-1);
                });

                it("should handle change", () => {
                    const props = createProps();
                    const tree = shallow(<Settings {...props} />);
                    const row = tree.find({ name });
                    const value = 8;
                    row.prop("onChange")(name, value);
                    expect(tree.state().config[name]).toBe(value);
                });

                it("should handle increment", () => {
                    const props = createProps();
                    const tree = shallow(<Settings {...props} />);
                    const row = tree.find({ name });
                    row.prop("onIncrement")(name);
                    expect(tree.state().config[name]).toBe(1);
                });
            });
        };

        suite("cards", "Player Deck Size");
        suite("epidemics", "Epidemic Count");
    });

    describe("infection deck", () => {
        it("should handle adding a city", () => {
            const props = createProps();
            const tree = shallow(<Settings {...props} />);
            const button = tree.find(SubHeaderButton);
            const city = "San Francisco";
            props.services.dialog.prompt.mockReturnValue(city);
            button.simulate("click");
            expect(tree.state("config").cities).toEqual({
                [city]: 1,
            });
        });

        it("should handle adding null", () => {
            const props = createProps();
            const tree = shallow(<Settings {...props} />);
            const button = tree.find(SubHeaderButton);
            props.services.dialog.prompt.mockReturnValue(null);
            button.simulate("click");
            expect(tree.state("config").cities).toEqual({});
        });

        it("should decrement a city", () => {
            const props = createProps();
            const tree = shallow(<Settings {...props} />);
            const city = "San Francisco";
            tree.setState({
                config: {
                    ...tree.state("config"),
                    cities: {
                        [city]: 2,
                    },
                },
            });
            tree.find({ name: city }).prop("onDecrement")(city);
            expect(tree.state("config").cities).toEqual({
                [city]: 1,
            });
        });

        it("should remove a city if the city falls to zero", () => {
            const props = createProps();
            const tree = shallow(<Settings {...props} />);
            const city = "San Francisco";
            tree.setState({
                config: {
                    ...tree.state("config"),
                    cities: {
                        [city]: 1,
                    },
                },
            });
            tree.find({ name: city }).prop("onDecrement")(city);
            expect(tree.state("config").cities).toEqual({});
        });

        it("should handle setting a city", () => {
            const props = createProps();
            const tree = shallow(<Settings {...props} />);
            const city = "San Francisco";
            tree.setState({
                config: {
                    ...tree.state("config"),
                    cities: {
                        [city]: 1,
                    },
                },
            });
            tree.find({ name: city }).prop("onChange")(city, 8);
            expect(tree.state("config").cities).toEqual({
                [city]: 8,
            });
        });

        it("should increment a city", () => {
            const props = createProps();
            const tree = shallow(<Settings {...props} />);
            const city = "San Francisco";
            tree.setState({
                config: {
                    ...tree.state("config"),
                    cities: {
                        [city]: 1,
                    },
                },
            });
            tree.find({ name: city }).prop("onIncrement")(city);
            expect(tree.state("config").cities).toEqual({
                [city]: 2,
            });
        });
    });

    describe("save", () => {
        describe("invalid state", () => {
            it("should reject if cards are zero", () => {
                const props = createProps();
                const tree = shallow(<Settings {...props} />);
                tree.find(BottomButton).simulate("click");
                expect(props.services.dialog.alert).toHaveBeenCalledWith("Invalid State");
                expect(props.onConfigure).not.toHaveBeenCalled();
            });

            it("should reject if epidemics are zero", () => {
                const props = createProps();
                const tree = shallow(<Settings {...props} />);
                tree.setState({
                    config: {
                        ...tree.state("config"),
                        cards: 1,
                    },
                });
                tree.find(BottomButton).simulate("click");
                expect(props.services.dialog.alert).toHaveBeenCalledWith("Invalid State");
                expect(props.onConfigure).not.toHaveBeenCalled();
            });

            it("should reject if cards are less than epidemics", () => {
                const props = createProps();
                const tree = shallow(<Settings {...props} />);
                tree.setState({
                    config: {
                        ...tree.state("config"),
                        cards: 1,
                        epidemics: 2,
                    },
                });
                tree.find(BottomButton).simulate("click");
                expect(props.services.dialog.alert).toHaveBeenCalledWith("Invalid State");
                expect(props.onConfigure).not.toHaveBeenCalled();
            });

            it("should reject if cities are empty", () => {
                const props = createProps();
                const tree = shallow(<Settings {...props} />);
                tree.setState({
                    config: {
                        ...tree.state("config"),
                        cards: 3,
                        epidemics: 2,
                    },
                });
                tree.find(BottomButton).simulate("click");
                expect(props.services.dialog.alert).toHaveBeenCalledWith("Invalid State");
                expect(props.onConfigure).not.toHaveBeenCalled();
            });
        });

        it("should not save on cancel", () => {
            const props = createProps();
            const tree = shallow(<Settings {...props} />);
            tree.setState({
                config: {
                    cards: 3,
                    cities: {
                        "San Francisco": 1,
                    },
                    epidemics: 2,
                },
            });
            props.services.dialog.confirm.mockReturnValue(false);
            tree.find(BottomButton).simulate("click");
            expect(props.services.dialog.confirm).toHaveBeenCalledWith("Reset Game State?");
            expect(props.onConfigure).not.toHaveBeenCalled();
        });

        it("should save", () => {
            const props = createProps();
            const tree = shallow(<Settings {...props} />);
            tree.setState({
                config: {
                    cards: 3,
                    cities: {
                        "San Francisco": 1,
                    },
                    epidemics: 2,
                },
            });
            props.services.dialog.confirm.mockReturnValue(true);
            tree.find(BottomButton).simulate("click");
            expect(props.services.dialog.confirm).toHaveBeenCalledWith("Reset Game State?");
            expect(props.onConfigure).toHaveBeenCalledWith(tree.state("config"));
        });
    });
});
