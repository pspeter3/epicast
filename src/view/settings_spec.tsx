import * as React from "react";
import { snapshotSuite } from "../util/snapshot";
import { Settings } from "./settings";

snapshotSuite("Settings", () => <Settings />);
