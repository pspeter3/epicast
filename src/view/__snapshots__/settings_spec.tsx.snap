// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings should match snapshot 1`] = `
Array [
  <section
    className="section"
  >
    <div
      className="row"
    >
      <p
        className="text"
      >
        Played Deck Size
      </p>
      <button
        className="icon-button"
      >
        <svg
          className="feather feather-minus-circle"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="8"
            x2="16"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
      <input
        className="numeric-input"
        type="number"
        value={60}
      />
      <button
        className="icon-button"
      >
        <svg
          className="feather feather-plus-circle"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="8"
            y2="16"
          />
          <line
            x1="8"
            x2="16"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
    </div>
    <div
      className="row"
    >
      <p
        className="text"
      >
        Epidemic Count
      </p>
      <button
        className="icon-button"
      >
        <svg
          className="feather feather-minus-circle"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="8"
            x2="16"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
      <input
        className="numeric-input"
        type="number"
        value={6}
      />
      <button
        className="icon-button"
      >
        <svg
          className="feather feather-plus-circle"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="8"
            y2="16"
          />
          <line
            x1="8"
            x2="16"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
    </div>
  </section>,
  <div
    className="row"
  >
    <h3
      className="subheader"
    >
      Infection Deck
    </h3>
    <button
      className="subheader-button"
    >
      Add City
    </button>
  </div>,
  <main
    className="main"
  >
    <section
      className="section"
    >
      <div
        className="row"
      >
        <p
          className="text"
        >
          San Francisco
        </p>
        <button
          className="icon-button"
        >
          <svg
            className="feather feather-minus-circle"
            fill="none"
            height="24"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="8"
              x2="16"
              y1="12"
              y2="12"
            />
          </svg>
        </button>
        <input
          className="numeric-input"
          type="number"
          value={2}
        />
        <button
          className="icon-button"
        >
          <svg
            className="feather feather-plus-circle"
            fill="none"
            height="24"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="16"
            />
            <line
              x1="8"
              x2="16"
              y1="12"
              y2="12"
            />
          </svg>
        </button>
      </div>
    </section>
  </main>,
  <button
    className="bottom-button"
  >
    Save
  </button>,
]
`;
